resqpy/property/_property_collection_get_attributes.py

```diff
@@ -0,0 +1,719 @@
+    elif indexable_element == 'nodes per cell':
+        shape_list = [support.nk, support.nj, support.ni, 2, 2,
+                      2]  # kp, jp, ip within each cell; todo: check RESQML shaping
+    elif indexable_element == 'nodes':
+        shape_list = _supporting_shape_grid_nodes(support)
+    return shape_list
+
+
+def _supporting_shape_grid_faces(direction, support):
+    assert direction is not None and direction.upper() in 'IJK'
+    axis = 'KJI'.index(direction.upper())
+    shape_list = [support.nk, support.nj, support.ni]
+    shape_list[axis] += 1  # note: properties for grid faces include outer faces
+    return shape_list
+
+
+def _supporting_shape_grid_nodes(support):
+    assert not support.k_gaps, 'indexable element of nodes not currently supported for grids with K gaps'
+    if support.has_split_coordinate_lines:
+        pillar_count = (support.nj + 1) * (support.ni + 1) + support.split_pillars_count
+        shape_list = [support.nk + 1, pillar_count]
+    else:
+        shape_list = [support.nk + 1, support.nj + 1, support.ni + 1]
+    return shape_list
+
+
+def _supporting_shape_wellboreframe(support, indexable_element):
+    if indexable_element is None or indexable_element == 'nodes':
+        shape_list = [support.node_count]
+    elif indexable_element == 'intervals':
+        shape_list = [support.node_count - 1]
+    return shape_list
+
+
+def _supporting_shape_blockedwell(support, indexable_element):
+    if indexable_element is None or indexable_element == 'intervals':
+        shape_list = [support.node_count - 1]  # all intervals, including unblocked
+    elif indexable_element == 'nodes':
+        shape_list = [support.node_count]
+    elif indexable_element == 'cells':
+        shape_list = [support.cell_count]  # ie. blocked intervals only
+    return shape_list
+
+
+def _supporting_shape_mesh(support, indexable_element):
+    if indexable_element is None or indexable_element == 'cells' or indexable_element == 'columns':
+        shape_list = [support.nj - 1, support.ni - 1]
+    elif indexable_element == 'nodes':
+        shape_list = [support.nj, support.ni]
+    return shape_list
+
+
+def _supporting_shape_gridconnectionset(support, indexable_element):
+    if indexable_element is None or indexable_element == 'faces':
+        shape_list = [support.count]
+    return shape_list
+
+
+def _supporting_shape_other(support, indexable_element):
+    if indexable_element is None or indexable_element == 'cells':
+        shape_list = [support.cell_count]
+    elif indexable_element == 'faces per cell':
+        support.cache_all_geometry_arrays()
+        shape_list = [len(support.faces_per_cell)]
+    return shape_list, support
+
+
+def _realizations_array_ref_get_r_extent(fill_missing, r_list):
+    if fill_missing:
+        r_extent = r_list[-1] + 1
+    else:
+        r_extent = len(r_list)
+    return r_extent
+
+
+def _add_part_to_dict_get_count_and_indexable(xml_node):
```

Comment:
This and the other _add_part_* functions might be better off living in _property_collection_add_part.py as they have to do with extracting data from xml during the adding of a property to the collection. This module would be better restricted to the functions that are inspecting the collection once it is constructed.